code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -458,16 +455,20 @@
 
 /datum/gas_reaction/nitrousformation/init_reqs()
 	requirements = list(
-		/datum/gas/oxygen = 10,
-		/datum/gas/nitrogen = 20,
-		/datum/gas/bz = 5,
+		/datum/gas/oxygen = MINIMUM_MOLE_COUNT,
+		/datum/gas/nitrogen = MINIMUM_MOLE_COUNT,
+		/datum/gas/bz = MINIMUM_MOLE_COUNT,
 		"MIN_TEMP" = N2O_FORMATION_MIN_TEMPERATURE,
 		"MAX_TEMP" = N2O_FORMATION_MAX_TEMPERATURE,
 	)
 
 /datum/gas_reaction/nitrousformation/react(datum/gas_mixture/air)
 	var/list/cached_gases = air.gases
-	var/heat_efficency = min(cached_gases[/datum/gas/oxygen][MOLES], cached_gases[/datum/gas/nitrogen][MOLES] * INVERSE(2))
+	var/temperature = air.temperature
+	var/minimal_mole_factor = min(cached_gases[/datum/gas/oxygen][MOLES], cached_gases[/datum/gas/nitrogen][MOLES] * INVERSE(2))
+	var/thermal_efficiency = 1 - ((temperature - N2O_FORMATION_OPTIMAL_TEMPERATURE) / N2O_FORMATION_TEMPERATURE_SCALE) ** 2
+	var/bz_efficiency = 1 - NUM_E ** -(thermal_efficiency * cached_gases[/datum/gas/bz][MOLES] / minimal_mole_factor)
+	var/heat_efficency = min(minimal_mole_factor * bz_efficiency, cached_gases[/datum/gas/oxygen][MOLES], cached_gases[/datum/gas/nitrogen][MOLES] * INVERSE(2))
```

Comment:
Jesus this is complex, what's going on here exactly?
Also why do you calculate minimal_mole_factor's value twice?

Reply:
thermal_efficiency is just a parabola that peaks at 1 when between minimum and maximum temperature.

The equation for bz_efficiency and other variables that look similar could be understood as an integral of e^-x, where x is whatever I write in the parenthesis. This would evaluate as progressing roughly 63.212% towards 100% every time x increases by 1. Since the slope of the equation is 1 when x is 0, it can be more easily understood as efficiency being slightly below x when x is a low value (I'd say below 0.2), so the efficiency would be almost 10% if x is 0.1.

I'm not sure if my explanation could be understood easily, but hopefully understanding that the efficiency would be near x when x is a low value, then having noticeable diminishing returns when x increases to high values (above 0.2) wouldn't be too problematic to maintain. This formula is very similar (exactly in probably one case) to what people would use if they wanted to calculate the chance of an event occuring where the same chance roll occurs multiple times, though I don't know if that has a name.

In this case for bz_efficiency formula, the efficiency rises as how much bz there is in relation to the minimum moles of other gases mapped with consumption ratios. Thermal efficiency dictates how efficient the bz would be. If thermal efficiency is at 100%, and there is 1 part bz per 5 part minimal moles, then you can assume that heat_efficiency would be almost 1/5th of minimal_moles.

Now for why minimal_moles gets calculated twice, the first one is used as part to calculate the production/consumption/heat_efficiency rates. The second one is redundant as mathematically this shouldn't go above minimum moles of any gas, but is used anyways to err on the side of caution, as with every gas reaction.